src/org/traccar/events/AlertEventHandler.java

```diff
@@ -26,11 +26,16 @@
 
     @Override
     protected Map<Event, Position> analyzePosition(Position position) {
-        Object alarm = position.getAttributes().get(Position.KEY_ALARM);
-        if (alarm != null) {
-            Event event = new Event(Event.TYPE_ALARM, position.getDeviceId(), position.getId());
-            event.set(Position.KEY_ALARM, (String) alarm);
-            return Collections.singletonMap(event, position);
+        Object alarms = position.getAttributes().get(Position.KEY_ALARM);
+        Map<Event, Position> eventMap = new LinkedHashMap<>();
+        if (alarms != null) {
+            String[] alarmList = alarms.toString().split(",");
```

Comment:
You would need to implement support for this on the web app side as well.

Reply:
This is how it looks without modifying the web app
![captura de pantalla de 2017-11-15 21-17-08](https://user-images.githubusercontent.com/8893928/32867230-bbdef6ca-ca4a-11e7-99dd-55e2b83b04ee.png)

Reply:
I would need to see what value is in the "alarm" field and then what it looks like in state, events and reports.

Reply:
![captura de pantalla de 2017-11-15 21-48-37](https://user-images.githubusercontent.com/8893928/32867984-c219e712-ca4e-11e7-9b74-6a513d25fe06.png)

Clicking both events at 21:45:42 in the report shows the same value for the "Alarm" field in the state frame.

Reply:
That's not this feature supposed to work. Alarm should contain key that indicates alarm type. Then web interface interprets this key using selected language.

Reply:
Documentation at https://www.traccar.org/documentation/computed-attributes/ says nothing about the accepted values.

Now, I think I see your point about the web interface. If the computed alarm is supposed to return a string with the alarm type then the web interface should be able to translate each alarm separately. With no modifications to the web interface it would not interpret multiple alarms correctly.

Reply:
I made a test with the master branch of the server and web app and it doesn't interpret the values for Alarm in the state pane as it does in the Event pane.

Reply:
Maybe it's not implemented there yet, but it definitely does it for event panel and report.

Reply:
I implemented it modifying the file web/app/view/State.js, but I don't know if that is the right place
![captura de pantalla de 2017-11-16 00-17-40](https://user-images.githubusercontent.com/8893928/32872164-fbeea1ca-ca63-11e7-97e8-f176397b8b4b.png)
 (please don't mind the meaningless definitions)